### PR TITLE
feat(role): add factorio

### PR DESCRIPTION
**Pull request by plinders**
# Description

Adds a a role to deploy a Factorio headless server. Will boot a fresh game with files at `/opt/factorio` if that folder is empty, otherwise will load the save if there is one. Additional configuration through `server-settings.json` as per the Factorio headless server [documentation](https://wiki.factorio.com/Multiplayer). 

[Docker image](https://hub.docker.com/r/goofball222/factorio).

Will try to make documentation soon.

# How Has This Been Tested?

- [X] Tested on my Sandbox install and tested connectivity to the game server

### CODE REVIEW

File: roles/factorio/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+  - "{{ factorio_paths_location }}"
+
+################################
+# Web
+################################
+
+factorio_web_subdomain: "{{ factorio_name }}"
+factorio_web_domain: "{{ user.domain }}"
+factorio_web_port: ""
+factorio_web_scheme: "https"
```
Comment (saltydk) [inline roles/factorio/defaults/main.yml:32]: Not needed.

File: roles/factorio/defaults/main.yml [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+factorio_web_serverstransport: "skipverify@file"
```
Comment (saltydk) [inline roles/factorio/defaults/main.yml:33]: Not needed.

File: roles/factorio/defaults/main.yml [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+factorio_web_url: "{{ 'https://' + (factorio_web_subdomain + '.' + factorio_web_domain
```
Comment (saltydk) [inline roles/factorio/defaults/main.yml:34]: Not needed.

File: roles/factorio/defaults/main.yml [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+                if (factorio_web_subdomain | length > 0)
+                else factorio_web_domain) }}"
+
+################################
+# DNS
+################################
+
+factorio_dns_record: "{{ factorio_web_subdomain }}"
+factorio_dns_zone: "{{ factorio_web_domain }}"
+factorio_dns_proxy: "{{ dns.proxied }}"
```
Comment (saltydk) [inline roles/factorio/defaults/main.yml:44]: This has to be set to
```yaml
factorio_dns_proxy: false
```

File: roles/factorio/defaults/main.yml [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+# Traefik
+################################
+
+factorio_traefik_sso_middleware: ""
+factorio_traefik_middleware_default: "{{ traefik_default_middleware }}"
+factorio_traefik_middleware_custom: ""
+factorio_traefik_certresolver: "{{ traefik_default_certresolver }}"
+factorio_traefik_enabled: true
+factorio_traefik_api_enabled: false
+factorio_traefik_api_endpoint: ""
```
Comment (saltydk) [inline roles/factorio/defaults/main.yml:56]: Entire Traefik section should be removed.

**Review by saltydk — CHANGES_REQUESTED** [review]

**Review by plinders — COMMENTED** [review]
Should all be resolved with commit (https://github.com/saltyorg/Sandbox/pull/288/commits/a91b331b18fb7485370311d76951c973bb8d1ef3)

**Pull request merged**